crates/sui-config/src/node.rs

```diff
@@ -163,13 +165,63 @@ impl NodeConfig {
     pub fn genesis(&self) -> Result<&genesis::Genesis> {
         self.genesis.genesis()
     }
+
+    #[allow(clippy::mutable_key_type)]
+    pub fn narwhal_worker_cache(&self) -> Result<narwhal_config::SharedWorkerCache> {
+        let genesis = self.genesis()?;
+        let consensus_config = self
+            .consensus_config()
+            .ok_or_else(|| anyhow!("cannot generate worker cache without ConsensusConfig"))?;
+        let workers = genesis
+            .validator_set()
+            .iter()
+            .map(|validator| {
+                let name = AuthorityPublicKey::from_bytes(validator.protocol_key().as_ref())
+                    .expect("Can't get protocol key");
+                let worker_address = if name != *self.protocol_key_pair().public() {
+                    validator.narwhal_worker_address.clone()
+                } else {
+                    // Use internal worker addresses for our own node if configured.
+                    consensus_config
+                        .internal_worker_address
+                        .as_ref()
+                        .unwrap_or(&validator.narwhal_worker_address)
+                        .clone()
+                };
+                let workers = [(
+                    0, // worker_id
```

Comment:
Not a concern of this PR, but since we are here the `worker_id` should also be configurable in the future and not hardcoded - but we can refactor this when we deal with the story of spinning up multiple workers.

Reply:
Yea it doesn't work at all for >1 worker right now so I figured that should all be done at once